app/main.py

```
@@ -1,14 +1,60 @@
-class Validator:
-    pass
+from abc import ABC, abstractmethod
 
 
-class Number:
-    pass
+class Validator(ABC):
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
 
+    def __get__(self, instance: object, owner: type) -> None:
+        return getattr(instance, self.protected_name)
 
-class OneOf:
-    pass
+    def __set__(self, instance: object, value: int) -> None:
+        self.validate(value)
+        setattr(instance, self.protected_name, value)
+
+    @abstractmethod
+    def validate(self, value: int) -> None:
+        pass
+
+
+class Number(Validator):
+    def __init__(self, min_value: int, max_value: int) -> None:
+        self.min_value = min_value
+        self.max_value = max_value
+
+    def validate(self, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Quantity should be integer.")
+        if not self.min_value <= value <= self.max_value:
+            raise ValueError(f"Quantity should not be less than "
+                             f"{self.min_value} and greater than "
+                             f"{self.max_value}.")
+
+
+class OneOf(Validator):
+    def __init__(self, *options) -> None:
+        self.options = options
+
+    def validate(self, value: int) -> bool:
+        if value in self.options:
+            return True
```

Issue: The validate method in OneOf returns True if the value is valid. According to the Validator base class and the Number validator, validate should not return anything (should just pass if valid and raise if invalid). Please remove the return True statement and use 'pass' instead for consistency.